card/mod/core/set/all/export.rb

```diff
@@ -1,30 +1,54 @@
 format :json do
-  before :export do
-    @export_depth = inherit(:export_depth).to_i + 1
-    @exported_keys = inherit(:exported_keys) || ::Set.new
-  end
-
+  # returns an array of Hashes (each in export_item view)
   view :export, cache: :never do
-    # avoid loops
-    return [] if @export_depth > max_export_depth || @exported_keys.include?(card.key)
-    @exported_keys << card.key
-    Array.wrap(render_atom).concat(render_export_items).flatten
+    exporting_uniques do
+      Array.wrap(render_export_item).concat(export_items_in_view(:export)).flatten
+    end
   end
 
   def max_export_depth
     Env.params[:max_export_depth].present? ? Env.params[:max_export_depth].to_i : 2
   end
 
-  before :export_items do
-    @exported_keys = inherit(:exported_keys) || ::Set.new
+  # returns an array of Hashes (each in export_item view)
+  view :export_items, cache: :never do
+    exporting_uniques do
+      export_items_in_view :export_item
```

Comment:
I'm wondering how expensive blocks are. I admit that this reads well but passing a block around only for initializing a instance variable seems like overkill. A simple method call would do the same job.